composer/algorithms/copypaste/copypaste.py

```diff
@@ -0,0 +1,403 @@
+    chosen (without replacement) from the source. Each instance goes through a set
+    of transformation and jittering, e.g., horizontal flipping, rescaling, and 
+    cropping. The resulting "jittered" instance is then checked for its area and 
+    transformed instances with an area smaller than the configured threshold. 
+    The resulting jittered instances are then pasted on a random position on 
+    the target. Same procedure is also applied on the masks of the source and 
+    the target. If a copied instances mask collides with an exisiting mask on 
+    the target, the copied instance's mask overlays the original mask on the 
+    target.
+
+    Args:
+        input (torch.Tensor): input tensor of shape ``(N, C, H, W)``.
+        target (torch.Tensor): target tensor of shape ``(N, H, W)``.   
+        configs (dict): dictionary containing the configurable hyperparameters.    
+
+    Returns:
+        out_images (torch.Tensor): batch of images after copypaste has been
+            applied.
+        out_masks (torch.Tensor): batch of corresponding masks after applying 
+        copypaste to them
+
+    Example:
+        .. testcode::
+
+            import torch
+            from composer.functional import copypaste_batch
+
+            N, C, H, W = 2, 3, 4, 5
+            num_classes = 10
+            configs = {
+                "p": 1.0,
+                "max_copied_instances": None,
+                "area_threshold": 100,
+                "padding_factor": 0.5,
+                "jitter_scale": (0.01, 0.99),
+                "jitter_ratio": (1.0, 1.0),
+                "p_flip": 1.0,
+                "bg_color": 0
+            }
+            X = torch.randn(N, C, H, W)
+            y = torch.randint(num_classes, size=(N, H, W))
+            out_images, out_masks = cutmix_batch(X, y, configs)
+    """
+    batch_idx = 0
+    out_images = torch.zeros_like(images)
+    out_masks = torch.zeros_like(masks)
+
+    assert images.size(dim=0) == masks.size(
+        dim=0), "Number of images and masks in the batch do not match!"
+    batch_size = images.size(dim=0)
+
+    while batch_idx < batch_size:
+        [i, j] = np.random.randint(0, high=batch_size, size=2)
+
+        num_instances = _count_instances(masks[i])
+        num_copied_instances = random.randint(0, num_instances)
+        if configs["max_copied_instances"] is not None:
+            num_copied_instances = min(
+                num_copied_instances, configs["max_copied_instances"])
+
+        src_instance_ids = _get_instance_ids(
+            masks[i], num_copied_instances, configs["bg_color"])
+
+        trg_image = images[j]
+        trg_mask = masks[j]
+
+        if random.uniform(0, 1) < configs["p"]:
+            for src_instance_id in src_instance_ids:
+                trg_image, trg_mask = _copypaste_instance(
+                    images[i], masks[i], trg_image, trg_mask, src_instance_id, configs)
```

Comment:
```suggestion
        if random.uniform(0, 1) < configs["p"]:
            for src_instance_id in src_instance_ids:
                trg_image, trg_mask = _copypaste_instance(
                    images[i], masks[i],  images[j], masks[j], src_instance_id, configs)
```

Reply:
Thanks so much for the feedback @eracah ! This change will break the logic as images are iteratively passed to _copypaste_instance()

Reply:
Oh my bad. didn't notice that